src/shared/fido2-util.c

```diff
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "fido2-util.h"
+#include "fileio.h"
+#include "libfido2-util.h"
+#include "memory-util.h"
+#include "random-util.h"
+
+int fido2_generate_salt(void **ret_salt, size_t *ret_size) {
```

Comment:
so if you touch this, I think it would make sense to switch this over to using "struct iovec" for this, with all the helpers from iovec-util.h, i.e. in particular `_cleanup_(iovec_done)` and things like that.

Reply:
In fact in https://github.com/systemd/systemd/pull/33276 one of the first commits adds `crypto_random_bytes_allocate_iovec()` which would implement pretty much exactly what you are implement here.